meson.build

```diff
@@ -112,6 +114,10 @@ c_cpp_cross_link_args = [
   '-nostdlib',
    # Only link static files
   '-static',
+  # Warn if we use COMMON
+  '-Wl,--warn-common',
+  # Warn if we include orphan sections
+  '-Wl,--orphan-handling=warn',
```

Comment:
Do we currently have anything that generates orphan sections? I'd probably suggest an error here if not.

Reply:
If I switch to using LLD, it errors claiming that `.symtab` and `.strtab` are orphan sections (the internal symbol and string tables ELF needs to actually work). I would like to be able to switch linkers as I investigate LLD, so I've chosen flags that work with both.

Reply:
If we end up needing DSECT for the DV logging (which LLD does not support), I'll turn this into `=error`.

Reply:
A quick google search turned up this workaround on LKML: https://lkml.org/lkml/2020/7/31/1042.

Reply:
Cool, that seems to work, and so i've added `-Wl,--fatal-warnings` so any issues brought up by the linker turn into errors.